### PR TITLE
do a console.warn if re-rendar is avoidable

### PR DESCRIPTION
**Pull request by FrancescoCioria**
image

@giogonzo this is a working draft, have a look

### CODE REVIEW

File: src/decorators/pure.js [hunk 1]
```diff
@@ -1,8 +1,17 @@
 import t from 'tcomb';
 import isReactComponent from '../isReactComponent';
 import debug from 'debug';
+import isEqual from 'lodash/isEqual';
```
Comment (FrancescoCioria) [inline src/decorators/pure.js]: this isn't part of dependencies yet. do we like `_isEqual`?

Comment (giogonzo) [inline src/decorators/pure.js]: yes it's fine

File: src/decorators/pure.js [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
 
+const testLogsMap = {};
 const log = debug('revenge:@pure');
+const logTest = (rootNodeID, displayName, key) => {
+  const component = `${displayName}-${rootNodeID}`;
+  if (!testLogsMap[component]) {
+    testLogsMap[component] = true;
+    console.warn(`Update avoidable! (component: ${displayName}, key: ${key}, rootNodeID: ${rootNodeID})`);
```
Comment (giogonzo) [inline src/decorators/pure.js]: this logs a single warning per component, but there could be more than one, given more re-renders right?

Comment (giogonzo) [inline src/decorators/pure.js]: meaning, probably `key` should be part of this map key too?

**Comment by giogonzo** [comment]
nice! I left some inline comments

**Comment by giogonzo** [comment]
In general (probably better to discuss this offline) what is the proposed workflow here?

I can imagine two different ones:
1. don't know what to optimize yet -> turn this on -> look for logs
2. I know what area/components I want too look into (e.g. using `Perf.printWasted` before) -> I'd like to turn these logs on for one or a few components to know exactly when and why they are rerendering

I kind of imagine the second workflow to be the main one (mainly because I expect 1 to be too verbose), what do you think?

In this case, I'd probably be better to not rely on a specific `NODE_ENV` and instead export a different function, e.g. `pureLog`

File: src/decorators/pure.js (hunk 2 above)
Comment (FrancescoCioria) [inline src/decorators/pure.js]: yes, `key` should be added

**Comment by FrancescoCioria** [comment]
yes, I'm also for having a different function (`pureLog`). should we import it in `pure` and globally activate it if `NODE_ENV === x` to support also the first case? (first case may be useful/common in the first phases of a global perf improvement)

**Comment by giogonzo** [comment]
Updated readme

**Pull request closed without merging**